demoapp/demo_app_functions.py

```diff
@@ -450,4 +466,479 @@ def update_csv(bill_num: str, title: str, summarized_bill: str, category: str, t
+
+    Args: 
+        response (str): Response string generated by the LLM. Ex: "Tags: ['tag_1', 'tag_2']"
+
+    Returns: 
+        list of either tags or categories. Ex: [tag_1, tag_2]
+
+    '''
+    
+    cats_str = response.split(':')[1]
+    
+    translate_dict = dict.fromkeys(map(ord, '-[]"'), None)
+    translate_dict[ord("'")] = None
+    cats_str = cats_str.translate(translate_dict)
+
+    return [i.strip() for i in cats_str.split(',')]
+
+def prepare_invoke_dict(bill_details: BillDetails) -> dict:
+
+    """
+    This function prepares the dict object that is used in chain.invoke function to call the LLM with prompt and 
+    required details. 
+
+    Args: 
+        bill_details (BillDetails): Object containing information about the bill - bill_text, bill_title, mgl_ref, commottee_info, mgl_names
+
+    Returns: 
+        dict object containing all the necessary keys and values required for invoke call. 
+
+    """
+    return {
+                "title": bill_details.bill_title, 
+                "context": [Document(page_content=x) for x in bill_details.bill_text],
+                "names": bill_details.mgl_names, 
+                "mgl_sections": [Document(page_content=x) for x in bill_details.mgl_ref],
+                "committee_info": bill_details.committee_info
+            }
+
+def get_query_for_summarization(bill_details: BillDetails, llm_call_type: str) -> str:
+
+    """
+        
+    This functions prepares a prompt based on the call type (small: No use of RAG, large: Use RAG) for bill summarization
+
+    Args: 
+        bill_details (BillDetails): Object containing information about the bill - bill_text, bill_title, mgl_ref, commottee_info, mgl_names
+        llm_call_type (str): This argument can take 2 values ("small": No use of RAG, "large": Use RAG) 
+
+    Returns: 
+        str: Query that includes task, context and instructions on summary generation
+
+    """
+
+    
+    if llm_call_type == 'large':
+        query = f""" Can you please explain what the following MA bill means to a regular resident without specialized knowledge? 
+                    Please provide a one paragraph summary in 4 sentences. Please be simple, direct and concise for the busy reader. Make bullet points if possible.
+                    Note that the bill refers to specific existing chapters and sections of the Mass General Laws. Use the information from those chapters and sections in your context to construct your summary
+                    Summarize the bill that reads as follows:\n{getattr(bill_details, "bill_text")}\n and is titled: {getattr(bill_details, "bill_title")}\n
+                    
+                    Use the information from those chapters and sections in your context as well as the corresponding {getattr(bill_details, "mgl_names")} of those chapter and sections to construct your summary, generating categories and tags.\n 
```

Comment:
```suggestion
                    Use the information from those chapters and sections in your context as well as the corresponding names of those chapter and sections to construct your summary, generating categories and tags.\n
                    
                     The chapter and section names are: {getattr(bill_details, "mgl_names")} 
```

Reply:
Can we double check if we actually have the section names? i.e. we want to provide something like,

```
Sectiomn 268A: "CONDUCT OF PUBLIC OFFICIALS AND EMPLOYEES"
```

Not just ,

````
"268A"
```